test/test_mechanize_form.rb

```diff
@@ -919,6 +919,34 @@ def test_has_field
     assert(form.has_field?('intarweb'))
   end
 
+  def test_add_file_upload
+    page = @mech.get("http://localhost/file_upload.html")
+    form = page.form_with(:name => 'value_test')
+
+    number_of_uploads = form.file_uploads.length
+
+    assert form.add_file_upload!('ham', '/etc/passwd', 'ohnoes', 'text/plain')
+    assert_equal(number_of_uploads + 1, form.file_uploads.length)
+
+    upload = form.file_upload_with(:name => 'ham')
+    assert upload
+    assert_equal '/etc/passwd', upload.file_name
+    assert_equal 'ohnoes', upload.file_data
+    assert_equal 'text/plain', upload.mime_type
+  end
```

Comment:
What happens if there is more than one with the same name? Seems like a janky way to store them. (not on you)